### ISSUE TITLE
Weird behaviour in Docker container

### ISSUE THREAD

**Issue by rieschl** [infection/infection#332]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.8.1
| Test Framework version | PHPUnit 7.1.5
| PHP version | 7.1.16
| Platform    | Docker `php:7.1`
| Github Repo | https://github.com/trinet-at/docker-infection

I use GitLab and the gitlab-ci task runner for testing my projects. The runner uses Docker to execute the tasks, having `php:7.1` as base image. When I added infection to my project I noticed a weird behavior when infection runs in the container. Locally everything works fine.
Infection complains that a required extension, in this test case the `intl` extension is not loaded, although it is.

To replicate the problem I created a small test scenario which uses the `Locale` class from `intl` from which the problem can be reproduced.

```
git clone https://github.com/trinet-at/docker-infection.git
cd docker-infection
composer install
docker build -t docker-infection .
```

Locally, infection runs fine:
`vendor/bin/infection`

If I run infection in the Docker container the error occurs (`vendor/bin/infection` is the default `CMD` in the image)

`docker run --rm -it -v $PWD:/app docker-infection`

However, running PHPUnit completes without a problem:
`docker run --rm -it -v $PWD:/app docker-infection vendor/bin/phpunit`

Also, running the `infection.phar` (installed as executable in `/usr/local/bin` in the image) succeeds:
`docker run --rm -it -v $PWD:/app docker-infection infection.phar`

The "solution" is to combine all of the `.ini` files of PHP into one `php.ini`.
There's a script in the repo which does exactly that and runs `vendor/bin/infection` afterwards:
`docker run --rm -it -v $PWD:/app docker-infection ./convert-ini.sh`

So it seems to me that infection somehow runs PHPunit in a non-standard way.

Does anybody have a solution for that without having to use the `.phar` file or combining PHP's `.ini` files? Maybe it's somehow related to issue #306.

Thanks!

**Comment by BackEndTea** [comment]
This is not fixed on master yet, but using #329 fixed it this issue.

**Comment by rieschl** [comment]
Ah, thanks for the quick reply!
